File: model/membership.go
package model

type MembershipLevel string

const (
	MembershipLevelFree MembershipLevel = "free"
	MembershipLevelVIP  MembershipLevel = "vip"
	MembershipLevelSVIP MembershipLevel = "svip"
)

// MembershipPlan 会员套餐配置。
type MembershipPlan struct {
	ID             string          `json:"id" gorm:"primaryKey"`
	Name           string          `json:"name"`
	Level          MembershipLevel `json:"level"`
	Description    string          `json:"description"`
	Price          int             `json:"price"`               // 价格，单位：分
	DurationDays   int             `json:"durationDays"`        // 会员有效天数
	CreditsGranted int             `json:"creditsGranted"`      // 赠送算力点数量
	Unlimited      bool            `json:"unlimited"`           // 会员期内是否不限算力点扣费
	PriorityQueue  bool            `json:"priorityQueue"`       // 是否优先队列
	Features       string          `json:"features" gorm:"type:text"` // JSON 文本，记录解锁的功能 key
	Enabled        bool            `json:"enabled"`
	Sort           int             `json:"sort"`
	CreatedAt      string          `json:"createdAt"`
	UpdatedAt      string          `json:"updatedAt"`
}

type MembershipPlanList struct {
	Items []MembershipPlan `json:"items"`
	Total int              `json:"total"`
}

type MembershipOrderStatus string

const (
	MembershipOrderStatusPending   MembershipOrderStatus = "pending"
	MembershipOrderStatusPaid      MembershipOrderStatus = "paid"
	MembershipOrderStatusCancelled MembershipOrderStatus = "cancelled"
)

type PaymentProvider string

const (
	PaymentProviderWechat PaymentProvider = "wechat"
	PaymentProviderAlipay PaymentProvider = "alipay"
	PaymentProviderMock   PaymentProvider = "mock"
)

// MembershipOrder 会员订单。
type MembershipOrder struct {
	ID              string                `json:"id" gorm:"primaryKey"`
	UserID          string                `json:"userId" gorm:"index"`
	PlanID          string                `json:"planId" gorm:"index"`
	PlanName        string                `json:"planName"`
	PlanLevel       MembershipLevel       `json:"planLevel"`
	Amount          int                   `json:"amount"` // 单位：分
	Status          MembershipOrderStatus `json:"status" gorm:"index"`
	PaymentProvider PaymentProvider       `json:"paymentProvider"`
	PaymentID       string                `json:"paymentId"`
	PayURL          string                `json:"payUrl"`
	PaidAt          string                `json:"paidAt"`
	ExpiresAt       string                `json:"expiresAt"`
	CreatedAt       string                `json:"createdAt"`
	UpdatedAt       string                `json:"updatedAt"`
}

type MembershipOrderList struct {
	Items []MembershipOrder `json:"items"`
	Total int               `json:"total"`
}

File: web/src/services/api/membership.ts
import { apiGet, apiPost, compactApiParams } from "@/services/api/request";
import type { AuthUser, MembershipLevel } from "@/services/api/auth";

export type PaymentProvider = "wechat" | "alipay" | "mock";

export type MembershipPlan = {
    id: string;
    name: string;
    level: MembershipLevel;
    description: string;
    price: number;
    durationDays: number;
    creditsGranted: number;
    unlimited: boolean;
    priorityQueue: boolean;
    features: string;
    enabled: boolean;
    sort: number;
    createdAt: string;
    updatedAt: string;
};

export type MembershipPlanListResponse = {
    items: MembershipPlan[];
    total: number;
};

export type MembershipOrderStatus = "pending" | "paid" | "cancelled";

export type MembershipOrder = {
    id: string;
    userId: string;
    planId: string;
    planName: string;
    planLevel: MembershipLevel;
    amount: number;
    status: MembershipOrderStatus;
    paymentProvider: PaymentProvider;
    paymentId: string;
    payUrl: string;
    paidAt: string;
    expiresAt: string;
    createdAt: string;
    updatedAt: string;
};

export type MembershipOrderListResponse = {
    items: MembershipOrder[];
    total: number;
};

export type MembershipQuery = {
    keyword?: string;
    page?: number;
    pageSize?: number;
};

export async function fetchMembershipPlans(token: string, query: MembershipQuery = {}) {
    return apiGet<MembershipPlanListResponse>("/api/v1/membership/plans", compactApiParams(query), token);
}

export async function fetchMyMembership(token: string) {
    return apiGet<AuthUser>("/api/v1/membership/me", undefined, token);
}

export async function fetchMyMembershipOrders(token: string, query: MembershipQuery = {}) {
    return apiGet<MembershipOrderListResponse>("/api/v1/membership/orders", compactApiParams(query), token);
}

export async function createMembershipOrder(token: string, planId: string, provider: PaymentProvider) {
    return apiPost<MembershipOrder>("/api/v1/membership/orders", { planId, provider }, token);
}

export async function cancelMembershipOrder(token: string, id: string) {
    return apiPost<boolean>(`/api/v1/membership/orders/${encodeURIComponent(id)}/cancel`, {}, token);
}

export async function mockPayMembershipOrder(token: string, id: string) {
    return apiPost<MembershipOrder>(`/api/v1/membership/orders/${encodeURIComponent(id)}/mock-pay`, {}, token);
}

export async function refreshMembershipOrderPay(token: string, id: string) {
    return apiPost<MembershipOrder>(`/api/v1/membership/orders/${encodeURIComponent(id)}/refresh-pay`, {}, token);
}
